compositor_render/src/renderer/scene.rs

```diff
@@ -0,0 +1,194 @@
+use std::{collections::HashMap, sync::Arc};
+
+use compositor_common::scene::{
+    InputId, InputSpec, NodeId, OutputId, Resolution, SceneSpec, ShaderParams, TransformNodeSpec,
+    TransformParams,
+};
+use log::error;
+
+use crate::{
+    registry::GetError,
+    transformations::{shader::Shader, web_renderer::WebRenderer},
+};
+
+use super::{
+    texture::{OutputTexture, Texture},
+    RenderCtx,
+};
+
+pub struct InputNode {}
+
+pub enum TransformNode {
+    Shader {
+        params: HashMap<String, ShaderParams>,
```

Comment:
Why do we keep the map here? We need to preserve the order of parameters passed by the user. Names of shader uniform struct fields don't matter and without parsing shader we are not able to pass them in the correct order.

Reply:
this is just a placeholder